### DISCUSSION TITLE
Module '"crypto"' has no exported member 'KeyObject'.

### DISCUSSION THREAD

**Discussion by nicob-29** [panva/jose#63] [Q&A]
Hello 

I try to use jose for a project .
We use TypeScript 
NodeJS version is 10.17. Upgrade NodeJs is not possible.

Compiling...
Using tsc v2.3.4
node_modules/jose/types/index.d.ts(6,10): error TS2305: Module '"crypto"' has no exported member 'KeyObject'.
node_modules/jose/types/index.d.ts(6,21): error TS2305: Module '"crypto"' has no exported member 'PrivateKeyInput'.
node_modules/jose/types/index.d.ts(6,38): error TS2305: Module '"crypto"' has no exported member 'PublicKeyInput'. 

'KeyObject' is not part of API of Crypto for nodeJs 10.17. What your recommendation except to upgrade nodeJS ? 

Thanks for your help 

Nicolas

**Comment by panva** [comment]
> 'KeyObject' is not part of API of Crypto for nodeJs 10.17. What your recommendation except to upgrade nodeJS ?

You don't need to upgrade node itself (altho yes that's the ultimate recommendation), but maybe you can upgrade just the [types](https://www.npmjs.com/package/@types/node).

https://github.com/panva/jose/issues/35#issuecomment-548235987
